Need help to understand the cause of NPE

Team,

In one of the environments, we encountered the below exception. This was a one off case amongst too many successful deployments. Thus, there is no reproducer. Needed your help in understanding what could be the cause of this so that we know steps to avoid / resolve this.

```
Caused by: java.lang.NullPointerException
	at oshi.software.os.linux.LinuxNetworkParams.getDomainName(LinuxNetworkParams.java:80) ~[oshi-core-6.0.0.jar!/:6.0.0]
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@
 * [#2095](https://github.com/oshi/oshi/pull/2095): Avoid using reserved identifiers as variable names - [@muhammetgumus](https://github.com/muhammetgumus)
 * [#2099](https://github.com/oshi/oshi/pull/2099): Remove useless public constructor for abstract class - [@victorjbassey](https://github.com/victorjbassey).
 * [#2124](https://github.com/oshi/oshi/pull/2124): Properly determine Apple Silicon frequency - [@dbwiddis](https://github.com/dbwiddis).
+* [#2133](https://github.com/oshi/oshi/pull/2133): Fix NPE for null canonical host name - [@dbwiddis](https://github.com/dbwiddis).
 
 # 6.2.0 (2022-06-26), 6.2.1 (2022-06-29)
 

--- a/oshi-core/src/main/java/oshi/software/os/linux/LinuxNetworkParams.java
+++ b/oshi-core/src/main/java/oshi/software/os/linux/LinuxNetworkParams.java
@@ -65,7 +65,7 @@ final class LinuxNetworkParams extends AbstractNetworkParams {
             try {
                 hostname = InetAddress.getLocalHost().getHostName();
             } catch (UnknownHostException e) {
-                LOG.error("Unknown host exception when getting address of local host: {}", e.getMessage());
+                LOG.warn("Unknown host exception when getting address of local host: {}", e.getMessage());
                 return "";
             }
             try (CloseablePointerByReference ptr = new CloseablePointerByReference()) {
@@ -76,10 +76,9 @@ final class LinuxNetworkParams extends AbstractNetworkParams {
                     }
                     return "";
                 }
-                Addrinfo info = new Addrinfo(ptr.getValue()); // NOSONAR
-                String canonname = info.ai_canonname.trim();
-                LIBC.freeaddrinfo(ptr.getValue());
-                return canonname;
+                try (Addrinfo info = new Addrinfo(ptr.getValue())) {
+                    return info.ai_canonname == null ? hostname : info.ai_canonname.trim();
+                }
             }
         }
     }

